OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceTests.cs

```diff
@@ -0,0 +1,227 @@
+        SetupFetchByFilterForAdmins(userId,  true, false, parentCATOTTGId, null, admin);
+
+        // Assert
+        Func<Task<SearchResult<WorkshopDto>>> result = () => sensitiveWorkshopService.FetchByFilterForAdmins();
+
+        // Act
+        result.Should().ThrowAsync<InvalidOperationException>()
+        .WithMessage($"Region admin with the specified ID: {userId} not found");
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_WhenFilterByCATOTTGId_ShouldReturnEntities()
+    {
+        // Arrange
+        var parentCATOTTGId = 10;
+        WorkshopFilterAdministration filter = new WorkshopFilterAdministration() { From = 3, Size = 10, CATOTTGId = 1 };
+        var resultExcpected = SetupFetchByFilterForAdmins(userId, false, false, parentCATOTTGId, filter);
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_WhenRoleMinistryAdmin_ShouldReturnEntities()
+    {
+        // Arrange
+        var institutionId = Guid.NewGuid();
+        var parentCATOTTGId = 1;
+        var admin = new MinistryAdminDto() { Id = userId, InstitutionId = institutionId };
+        WorkshopFilterAdministration filter = new WorkshopFilterAdministration() { Size = 8, CATOTTGId = 1, From = 2, SearchString = "key word" };
+        var resultExcpected = SetupFetchByFilterForAdmins(userId, false, true, parentCATOTTGId, filter, null, admin);
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        resultExcpected.Should().BeEquivalentTo(result);
+    }
+
+    #endregion
+
+    #region With
+
+    private SearchResult<WorkshopDto> SetupFetchByFilterForAdmins(
+        string userId, bool isRegionAdmin, bool isMinistryAdmin, long parentCATOTTGId, WorkshopFilterAdministration filter, RegionAdminDto adminRegion = null, MinistryAdminDto adminMinistry = null)
+    {
+        var workshops = WithListOfWorkshops().ToList();
+        var workshopsDto = WithListOfWorkshopsDto().ToList();
+        IEnumerable<long> subSettlementsIds = new List<long>() { 1, 2, 3 };
+        SetUpCurrentUserService(userId, isRegionAdmin, isMinistryAdmin);
+        SetupRegionAdminService(userId, adminRegion);
+        SetupMinistryAdminService(userId, adminMinistry);
+        SetUpCodeficatorService(subSettlementsIds, parentCATOTTGId);
+        SetUpWorkshopsRepository(workshops, filter);
+        SetUpMapper(workshopsDto, workshops);
+        var resultExcpected = new SearchResult<WorkshopDto> { TotalAmount = workshopsDto.Count, Entities = workshopsDto };
+        return resultExcpected;
+    }
+
+    private IEnumerable<Workshop> WithListOfWorkshops()
+    {
+        var workshops = WorkshopGenerator.Generate(5);
+        return workshops;
+    }
```

Comment:
(Design) What was the intention to extract this (and next) methods. They are one-liners, called ony once and can be inlined in that single call place.

Reply:
Fixed